lib/geom/eli/geom_eli.c

```diff
@@ -1365,6 +1365,12 @@ eli_setkey_detached(struct gctl_req *req, const char *prov,
 		return;
 	}
 
+	/*
+	 * The previous eli_genkey() set cached_passphrase, we do not want
+	 * to use that for the new passphrase so always prompt for it
+	 */
+	explicit_bzero(cached_passphrase, sizeof(cached_passphrase));
```

Comment:
In this case a plain`memset` is probably ok.  Typically one uses `explicit_bzero` when a secret is no longer needed and you want to prevent it leaking into another context, not to invalidate a cached secret that is intended normally to be reused.  For this case I think `explicit_bzero` might be a bit confusing / misleading as it is not serving it's normal role.

Reply:
aye, that's good feedback. will fix

Reply:
okay, so, after consulting the rest of the code, there's not a single `memset`, or even `bzero` in all of lib/geom/eli/

Reply:
Older BSD code uses `bzero`, but newer code uses `memset` since `memset` is POSIX and `bzero` is not.  (`bzero` in the kernel is just mapped to `memset` via a macro.)